chain/network/src/types.rs

```diff
@@ -256,14 +241,91 @@ impl From<Handshake> for network_proto::Handshake {
         network_proto::Handshake {
             version: handshake.version,
             peer_id: handshake.peer_id.into(),
-            account_id,
             listen_port,
             chain_info: SingularPtrField::some(handshake.chain_info.into()),
             ..Default::default()
         }
     }
 }
 
+/// Account announcement information
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct AnnounceAccount {
+    /// AccountId to be announced
+    pub account_id: AccountId,
```

Comment:
So to maintain security, we need to have next information:
<account_id, validator_peer_id> + <peer_id sent to> signed with validator public key

when route from one existing peer to next peer, need to attach what have received before and then <peer_id sent to> + signature of the current peer_id

The problem is that self referential data structures don't work in Rust very well.

So data structure should be something like 
```
struct AnnounceAccountRoute {
   peer_id_sender: PeerId,
   peer_id_receiver: PeerId,
}

struct AnnounceAccount {
    account_id: AccountId
    route: Vec<AnnounceAccountRoute>,
    signatures: Vec<Signature>
}
```

and then methods to dissect route into parts and check signatures.
E.g. `fn prefix() -> AnnounceAccount` returns same AnnounceAccount but removing one item from route and signature - you can then check it's hash matches the last signature. or something like this.

Reply:
Oh, and when you send it, you wrap existing with new route that records to whom you send and sign that.

Reply:
But the signers are account_id holders, not peer_id holders. On the other hand the route is created among peer_id not account_id, do we want full mapping between account_id, peer_id while making the route. 

Remember that we can be connected to some peer and not have its account_id, in such case we can't verify such signature.

Reply:
But the signers are account_id holders, not peer_id holders. On the other hand the route is created among peer_id not account_id, do we want full mapping between account_id, peer_id while making the route.

Remember that we can be connected to some peer and not have its account_id, in such case we can't verify such signature.